# ã‚¤ãƒ³ãƒ—ãƒƒãƒˆï¼ˆInputï¼›å…¥åŠ›ï¼‰
extends Node


# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼
# ãƒãƒ¼ãƒ‰ãƒ»ãƒ‘ã‚¹é–¢é€£
# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼

# çŒ¿å–å¾—
func monkey():
	return $"../ğŸµMonkey"


# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼
# å…¥åŠ›
# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼

# ãƒ†ã‚­ã‚¹ãƒˆãƒœãƒƒã‚¯ã‚¹ãªã©ã«ãƒ•ã‚©ãƒ¼ã‚«ã‚¹ãŒç„¡ã„ã¨ãã®å…¥åŠ›ã‚’ã¨ã«ã‹ãæ‹¾ã†
func _unhandled_input(event):
	# ã‚­ãƒ¼ã‚³ãƒ³ãƒ•ã‚£ã‚°ã§å¿…è¦ã§ã™
	self.monkey().key_config_node().on_unhandled_input(event)


# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼
# ã‚¤ãƒ™ãƒ³ãƒˆãƒ»ãƒãƒ³ãƒ‰ãƒ©ãƒ¼
# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼

# ã‚­ãƒ¼ãƒ»ã‚³ãƒ³ãƒ•ã‚£ã‚°é–‹å§‹æ™‚
func on_key_config_entered():
	print("ï¼»ã‚­ãƒ¼ã‚³ãƒ³ãƒ•ã‚£ã‚°ï¼½ã€€é–‹å§‹")
	
	# èƒŒæ™¯è¡¨ç¤º
	self.monkey().of_staff().illustrator_node().visible = true
	self.monkey().of_staff().programmer().owner_node().images.find_node(
		str(self.monkey().of_staff().config_node().key_config_background_image_name)
	).visible = true


# ã‚­ãƒ¼ãƒ»ã‚³ãƒ³ãƒ•ã‚£ã‚°çµ‚äº†æ™‚
func on_key_config_exited():
	print("ï¼»ã‚­ãƒ¼ã‚³ãƒ³ãƒ•ã‚£ã‚°ï¼½ã€€å®Œäº†")

	# èƒŒæ™¯æ¶ˆå»
	self.monkey().of_staff().programmer().owner_node().images.find_node(
		str(self.monkey().of_staff().config_node().key_config_background_image_name)
	).visible = false
